Mathlib/Analysis/InnerProductSpace/Projection.lean

```diff
@@ -475,112 +476,197 @@ theorem orthogonalProjectionFn_eq (v : E) :
     K.orthogonalProjectionFn v = (K.orthogonalProjection v : E) :=
   rfl
 
+/-- The orthogonal projection onto a subspace as a map from the full space to itself,
+as opposed to `Submodule.orthogonalProjection`, which maps into the subtype. This
+version is important as it satisfies `IsStarProjection`. -/
+def starProjection (U : Submodule 𝕜 E) [U.HasOrthogonalProjection] :
+    E →L[𝕜] E := U.subtypeL ∘L U.orthogonalProjection
+
+lemma starProjection_apply (U : Submodule 𝕜 E) [U.HasOrthogonalProjection] (v : E) :
+    U.starProjection v = U.orthogonalProjection v := rfl
```

Comment:
It's fine to keep this lemma around, but I strongly suspect we want the reverse direction with `simp`:
```lean
lemma coe_orthogonalProjection_apply (U : Submodule 𝕜 E) [U.HasOrthogonalProjection] (v : E) :
     U.orthogonalProjection v = U.starProjection v := rfl
```